floodwatch/src/js/components/Profile.js

```diff
@@ -0,0 +1,198 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Row, Col, Button, ListGroup, ListGroupItem } from 'react-bootstrap';
+import _ from 'lodash'
+import Filters from '../../stubbed_data/filter_response.json';
+import DemographicKeys from '../../stubbed_data/demographic_keys.json';
+import type {DemographicDictionary} from './FindInDemographics'
+
+import {FWApiClient} from '../api/api';
+import type {PersonDemographicRequest} from '../api/types';
+
+type StateType = {
+  userData?: PersonDemographicRequest,
+  toPick: Array<string>,
+  curStatus: string
+};
+
+export class Profile extends Component {
+  render() {
+    return (
+      <Row style={{backgroundColor:'white', borderRadius:'10px', padding:'20px'}}>
+        <Col xs={12}>
+          <h3>My Profile</h3>
+          <p>Donate your data to help us discover discriminatory patterns in advertising, and reverse the power relationship between people and advertisers.</p>
+          <p>Wondering why your demographic data matters? <Button bsSize="xsmall">Read more</Button></p>
+        </Col>
+        <Col xs={12}>
+        <DemographicContainer/>
+        <AccountOptionsContainer/>
+        </Col>
+      </Row>
+    )   
+  }
+
+}
+
+export class DemographicContainer extends Component {
+  state: StateType
+
+  constructor() {
+    super();
+    this.state = {
+      toPick: ['birth_year', 'twofishes_id', 'demographic_ids'],
+      curStatus: ''
+    }
+  }
+
+  async componentDidMount() {
+    const UserData = await FWApiClient.get().getCurrentPerson()
+    let filteredUserData = _.pick(UserData, this.state.toPick)
+
+    this.setState({
+      userData: filteredUserData
+    })
+  }
+
+  handleClick(checked: boolean, event: any): void {
+    let demo = event.target.textContent;
+    let demo_id = _.find(DemographicKeys.demographic_keys, (o: DemographicDictionary) => {
+      return o.name == demo
+    })
+
+    if (event.target.name != 'age' && event.target.name != 'country_code') {
+      console.log(checked)
+      if (checked) {
+        this.addToDemographicIds(demo_id.id)
+      } else {
+        this.removeFromDemographicIds(demo_id.id)
+      }
+    }
+  }
+
+  async updateUserInfo() {
+    try {
+      const reply = await FWApiClient.get().updatePersonDemographics(this.state.userData)  
+      this.setState({curStatus: 'success'})
+    } catch (e) {
+      console.error(e)
+      this.setState({curStatus: 'error'})
+    }
+  }
+
+
+
+  addToDemographicIds(id: number): void {
+    let userData = _.cloneDeep(this.state.userData);
+    userData.demographic_ids.push(id)
+    this.setState({
+      userData: userData,
+      curStatus: (this.state.curStatus != 'error') ? '' : 'error'
```

Comment:
Maybe you want a `clearError()` method that handles this so it's not in a bunch of places?